// dbvi/index.js
exports.products=products=[{
  id:1,
  product : 'Piano Cottura a induzione',
  description : 'I piani cottura a induzione hanno il vantaggio di trasferire l’energia direttamente alle pentole, in modo da disperdere pochissimo calore, e sono molto rapidi, così passi meno tempo in cucina e abbassi la bolletta dell’elettricità.',
  amount:10
},
{
  id: 2,
  product : 'Forno Termoventilato',
  description : 'Nei forni termoventilati il calore si diffonde velocemente e uniformemente, così risparmi tempo ed energia quando cucini',
  amount:12
},
{
  id: 3,
  product : 'Lavastoviglie',
  description : 'Le nostre lavastoviglie sono dotate di programmi e funzionalità che consentono di risparmiare energia e di utilizzare la corretta quantità d’acqua per lavaggi ottimali. Un risparmio per l’ambiente e per la bolletta.',
  amount:20
},
{
  id:4,
  product : 'Frigorifero',
  description : 'I nostri frigoriferi e congelatori hanno funzioni intelligenti che mantengono gli alimenti freschi più a lungo, così riduci gli scarti e la frequenza con cui fai la spesa.',
  amount:15
},
{
  id:5,
  product : 'Lavatrice',
  description : 'Le nostre lavatrici sono progettate per prendersi cura di tutti i tuoi capi, anche i più delicati e quelli a cui tieni di più. Grazie alle funzioni semplici da usare e ai programmi speciali, fare il bucato è un gioco da ragazzi.',
  amount:23
}];

exports.shopping=shopping=[];
var messageDelete={};
var messageEdit={};
var messageBuy={};

exports.add = function (product,description,amount) {

  var newID = parseInt(products[products.length-1].id)+1;
  if(product==undefined ){
    product="undefined";
  }
  if(description==undefined){
    description="undefined";
  }
  if(isNaN(amount)){
    amount=0;
  }

  products.push({
    "id": newID,
    "product" : product,
    "description" : description,
    "amount": amount
  });
};


exports.del = function(id){
  id = parseInt(id);
  check=false;
  for(var i=0;i<products.length;i++){
    if(id==products[i].id){
      products.splice(i,1);
      check=true;
      messageDelete={message:"the product was deleted successfully"};
      return messageDelete;
    }
  }
  if(check==false){
    messageDelete={message:"the id in not present in the database"};
    return messageDelete;
  }
};

exports.buy = function(id,buyer){
  check=false;
  if(buyer==undefined){
    buyer="undefined";
  }

  for(var i=0;i<products.length;i++){
    if(id==products[i].id){
      if(products[i].amount===0){
      messageBuy={message:"the product was finished"};
      return messageBuy;
      }
      check=true;
      shopping.push({
        id:products[i].id,
        date:new Date(),
        buyer:buyer,
      });
      products[i].amount--;
      messageBuy={message:"the product was buyed"};
      return messageBuy;
    }

  }
  if(check==false){
    messageBuy={message:"the product not exist in the database"};
    return messageBuy;
  }

};

exports.edit = function(id,name, description,amount){
  check=false;
  for(var i=0;i<products.length;i++){
      if(id==products[i].id){
        check=true;
        if(name==undefined){
          name=products[i].product;
        }
        if(description==undefined){
          description=products[i].description;
        }
        if(isNaN(amount)){
          amount=products[i].amount;
        }
          products[i]={
          id:products[i].id,
          product:name,
          description:description,
          amount:amount
        };
        messageEdit={message:"the product was edited successfully"};
        return messageEdit;
      }
    }
    if(check==false){
      messageEdit={message:"the id is not present in the database"};
      return messageEdit;
    }
};

exports.search = function (buyer) {
  var arraySearch=[];
  for(var i=0;i<shopping.length;i++){
    if(buyer==shopping[i].buyer){
      arraySearch.push(shopping[i]);
    }
  }
  return arraySearch;
};
